mat64/io.go

```diff
@@ -81,18 +87,24 @@ func (m *Dense) UnmarshalBinary(data []byte) error {
 		panic("mat64: unmarshal into non-zero matrix")
 	}
 
+	var tmp [8]byte
 	buf := bytes.NewReader(data)
-	var rows int64
-	err := binary.Read(buf, binary.LittleEndian, &rows)
+	_, err := buf.Read(tmp[:])
```

Comment:
The error logic is not quite right in readFull, it should be a trimmed version of io.ReadFull without the min check.